src/Mailjet/Client.php

```diff
@@ -119,9 +138,16 @@ public function get($resource, $args = [])
      * @param array $args     Request arguments
      * @return Response
      */
-    public function put($resource, $args = [])
+    public function put($resource, array $args = [], array $options = [])
     {
-        return $this->_call('PUT', $resource[0], $resource[1], $args);
+        if (!empty($options)) {
+            $this->setOptions($options, $resource);
+        }
+        $result = $this->_call('PUT', $resource[0], $resource[1], $args);
+        if (!empty($settings['changed'])) {
```

Comment:
Empty allow me to check if the variable exist and if it true or false. if the variable is false, empty will return "true". Or, we only want to reset the settings if they changed before. So, we only enter if empty is false (which mean if the variable exist AND is equal at true)